### PR TITLE
[minor][attention] Better invariance and equivariance checks

### PR DESCRIPTION
**Pull request by blefaudeux**
## What does this PR do?
For completeness' sake, we were testing equivariance but not invariance. Credits to Francois Fleuret for the [heads up to all](https://twitter.com/francoisfleuret/status/1539622884766896128)

## Before submitting
- [x] Did you have fun?
  - Make sure you had fun coding 🙃
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/xformers/blob/master/CONTRIBUTING.md)?
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [x] N/A
- [ ] Did you make sure to update the docs?
  - [ ] N/A
- [x] Did you write any new necessary tests?
  - [ ] N/A
- [ ] Did you update the [changelog](https://github.com/facebookresearch/xformers/blob/master/CHANGELOG.md)? (if needed)
  - [x] N/A

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

File: tests/test_attentions.py [hunk 1]
```diff
@@ -127,27 +127,33 @@ def test_order_invariance(
     ):
         pytest.skip(f"{attention_name} requires squared sequence lengths")
 
-    # Check that a shuffled input produces the same results
```
Comment (blefaudeux) [inline tests/test_attentions.py:130]: the comments were flipped here, probably botched copy pasta

File: tests/test_attentions.py [hunk 2]
```diff
@@ -127,27 +127,33 @@ def test_order_invariance(
...
+    # Check that we can pass a smaller sequence
+    seqs = (
+        [SEQ, SEQ // 2]
+        if not multi_head.attention.requires_same_k_q_dimensions
+        else [SEQ]
+    )
 
     for seq in seqs:
-        # Check that we can pass a smaller sequence
+        # Check that the attention is invariant to a permutation of K, V
```
Comment (blefaudeux) [inline tests/test_attentions.py:138]: properly testing invariance and equivariance here

File: tests/test_attentions.py [hunk 3]
```diff
@@ -127,27 +127,33 @@ def test_order_invariance(
...
 
-        # Test the non-self-attention codepath
-        att = multi_head(inputs, inputs_shuffled, inputs)
-
         # Check that dropout actually drops some values
         if attn_dropout > 0:
+            att_1 = multi_head(inputs, inputs_shuffled, inputs)
             att_2 = multi_head(inputs, inputs_shuffled, inputs)
-            assert (att != att_2).any()
+            assert (att_1 != att_2).any()
```
Comment (blefaudeux) [inline tests/test_attentions.py:156]: this could randomly fail ("could"), but the seed is set and actually does not fail (not changed by this PR)

**Review by dianaml0 — APPROVED** [review]
Nice!

**Pull request merged**